### DISCUSSION TITLE
Keeping an RBC in place

### DISCUSSION THREAD

**Discussion by mojtabaAmiraslanpour** [UvaCsl/HemoCell#47] [General discussion]
Hello all,

I am trying to keep a single red blood cell in place artificially and exert shear flow on it for some reason. However, I cannot find the location of the nodes for the red blood cell and move it back to place when it changes location. I could go this far:

```
void HemoCellFields::HemoAdvanceParticles::processGenericBlocks(Box3D domain, std::vector<AtomicBlock3D*> blocks) {
    dynamic_cast<HEMOCELL_PARTICLE_FIELD*>(blocks[0])->advanceParticles();
}
void HemoCellFields::advanceParticles() {
  global.statistics.getCurrent()["advanceParticles"].start();
    
  vector<MultiBlock3D*> wrapper;
  wrapper.push_back(immersedParticles);
  applyProcessingFunctional(new HemoAdvanceParticles(),immersedParticles->getBoundingBox(),wrapper);

  global.statistics.getCurrent().stop();
}
```

But could not go further and find how to have access to the position of each node. Can you help me with that?

Best,
Moj

**Comment by gzavo** [comment]
The a look at [here in the repo](https://github.com/UvaCsl/HemoCell/blob/cf106f34aa015ee28d8de32ccd440ff3fb5abb16/core/hemoCellParticleField.cpp#L566).
The fundamental particle class is from Palabos, so the implementation of 'advance' is there. Note:this is not a complete cell, just a vertex of the membrane.
If you want to keep the particle in place, it should be enough to skip calling 'advanceParticles'. Then their position is not updated, their velocity will be 0, and start to behave like a solid object.
